common/app/com/commercetools/sunrise/common/sessions/TypedSessionStrategy.java

```diff
@@ -0,0 +1,35 @@
+package com.commercetools.sunrise.common.sessions;
+
+import com.google.inject.ImplementedBy;
+
+import java.util.Optional;
+
+/**
+ * Represents a strategy used to read and write objects into the user's session.
+ */
+@ImplementedBy(SerializableSessionCookieStrategy.class)
+public interface TypedSessionStrategy extends SessionStrategy {
```

Comment:
I didn't like `ObjectSessionStrategy` but couldn't find a replacement. I'm testing my luck with `TypedSessionStrategy`, as defining which type are you storing is important here. But I don't really like it either, it sounds as String is not a type. Help wanted here :(